### PR TITLE
Generate uniform dataset from Lichess data

### PR DESCRIPTION
**Pull request by owenjaques**
This PR adds one new notebook which reads the data from Lichess and creates a uniform dataset of board positions evaluated by stockfish. It also provides 125,000 samples that were generated from running that notebook in a training, testing, and validation sets (80%, 10%, 10%). 

The samples are in the form of FEN strings which are used to represent the board and they are labelled with floats in the range [0,1] which represent the win/draw/loss ration given by stockfish. To train a model using this dataset some pre-processing is likely required as the inputs are FEN strings, however, the Python chess module can be given a FEN string to initialize a board, which can then be used to create any board representation you like. I chose to do it this way as I would now like to experiment with giving neural networks different inputs, so, instead of filtering the Lichess data each time to create a uniform dataset, it is much easier to filter it once here then pre-process the dataset as required.

### CODE REVIEW

**Comment by Kayisn** [comment]
Nicely done! Looking forward to training a model with this data. Maybe I'll try to get the neural network copilot helped me write with this!

File: pre_processing/uniform_dataset.ipynb
Comment (Kayisn) [inline pre_processing/uniform_dataset.ipynb:1]: I like how you did the bins, much more efficient than what I was doing. 100,000 games should be a good starting point. :)

**Review by Kayisn — APPROVED** [review]

**Review by btfcopes — APPROVED** [review]
looks good

File: pre_processing/uniform_dataset.ipynb
Comment (owenjaques) [reply pre_processing/uniform_dataset.ipynb:1]: Thank you :)
